### ISSUE TITLE
Product Cart cannot be displayed if Welcome Message contains single quotes

### ISSUE THREAD

**Issue by bh-ref** [magento/magento2#5003]
## Steps to reproduce
1. Install Magento 2.1.0-rc2.
2. Add a product to the cart.
3. Switch to another language (storeview) which has a welcome message that contains a single quote. In my case, the default welcome message of my french storeview is `Email d'accueil par défaut`.
## Expected result
1. The same products can be seen in the cart when I click on the cart icon.
## Actual result
1. No products can be seen in the cart.
2. Chrome dev tools show the following error message:

```
knockout.js:2624 Uncaught SyntaxError: Unable to process binding "ifnot: function (){return customer().fullname }"
Message: Unable to parse bindings.
Bindings value: html:'Email d'accueil par défaut'
Message: Unexpected identifiercreateBindingsStringEvaluator @ knockout.js:2624createBindingsStringEvaluatorViaCache @ knockout.js:2615ko.utils.extend.parseBindingsString @ knockout.js:2601ko.utils.extend.getBindingAccessors @ knockout.js:2583ko.dependentObservable.disposeWhenNodeIsRemoved @ knockout.js:2931evaluateImmediate @ knockout.js:1737ko.computed.ko.dependentObservable @ knockout.js:1946applyBindingsToNodeInternal @ knockout.js:2929applyBindingsToNodeAndDescendantsInternal @ knockout.js:2854applyBindingsToDescendantsInternal @ knockout.js:2836ko.applyBindingsToDescendants @ knockout.js:3052ko.computed.disposeWhenNodeIsRemoved @ knockout.js:4004evaluateImmediate @ knockout.js:1737ko.computed.ko.dependentObservable @ knockout.js:1946ko.bindingHandlers.(anonymous function).init @ knockout.js:3988(anonymous function) @ knockout.js:2989ko.computedContext.ko.dependencyDetection.ignore @ knockout.js:1249(anonymous function) @ knockout.js:2988ko.utils.arrayForEach @ knockout.js:151applyBindingsToNodeInternal @ knockout.js:2974applyBindingsToNodeAndDescendantsInternal @ knockout.js:2854applyBindingsToDescendantsInternal @ knockout.js:2836ko.applyBindingsToDescendants @ knockout.js:3052applyComponents @ scope.js:30Registry._resolveRequest @ registry.js:413Registry._updateRequests @ registry.js:395later @ underscore.js:821
```

To reproduce this error, you might as well go to `app/code/MagentoTheme/view/frontend/templates/html/header.phtml` and replace line 12 with this: 

``` php
$welcomeMessage = "t'est";
```

and clean the cache and reload the page.

**Comment by cpartica** [comment]
created internal issue MAGETWO-54590

**Comment by davidbilly** [comment]
@bh-ref i have same issue : https://github.com/magento/magento2/issues/5426#issuecomment-231778311 

This fixed after remove ' from welcome message. Thanks!

**Comment by magento-engcom-team** [comment]
@bh-ref, thank you for your report.
The issue is already fixed in develop branch, 2.2.0

**Comment by ChameleonDevil** [comment]
> @bh-ref, thank you for your report.
> The issue is already fixed in develop branch, 2.2.0

I don't think this was fixed completely.  I have installed a new **M2.3.5** installation where I have this same error - when trying to debug customer() at the knockout section of the header template at the welcome message it returns `{}` empty object.

_I am getting the exact same error message as in issue #5426_  which is related to this Welcome message.

https://github.com/magento/magento2/issues/5426

Difference here is that I do not have products in this database yet.  Can someone please point me to where customer() gets populated, I need to figure out why customer() is empty object at that point.

**Comment by ChameleonDevil** [comment]
Ok so this issue is related to how knockout seems to not correctly handle (') single quote characters in some theme templates.

If your template has this:


<span data-bind="html:'<?= $block->escapeHtml($welcomeMessage) ?>'"></>


Change to this:


<span data-bind='html:"<?= $block->escapeHtml($welcomeMessage) ?>"'></>


Notice I swapped the single quote and double quote positions such that the double quotes are inside, so that in effect does this: 

`' " ' " '`
Instead of: 
`" ' ' ' "`

I am positive **anywhere (other templates)** where knockout data-bind is used with PHP `escapeHtml()` or where quote characters are present will have this issue.
